lib/clerk/jwks_cache.rb

```diff
@@ -0,0 +1,25 @@
+class JWKSCache
+  def initialize(lifetime)
+    @lifetime = lifetime
+    @jwks = nil
+    @fetched_at = nil
+    @lock = Concurrent::ReadWriteLock.new
+  end
+
+  def fetch(sdk, force_refresh: false)
+    should_refresh = @lock.with_read_lock do
+      force_refresh || @jwks.nil? || (@fetched_at && ((Time.now.to_i-@fetched_at) > @lifetime))
```

Comment:
:upside_down_face: Perhaps cast `@fetched_at.to_i` to avoid any unpleasant errors?

Reply:
Given that this is a tightly-controlled/internal instance var, I'd prefer not to, since this could result in hard-to-diagnose bugs otherwise (`""` and `nil` both cast to `0`). If we screw this up somehow, I'd prefer for it to fail loud so that we catch this earlier.

We also have test for this path.